hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -196,6 +196,11 @@ public void initialize(URI uri, Configuration configuration)
 
     final AbfsConfiguration abfsConfiguration = abfsStore
         .getAbfsConfiguration();
+
+    // Ensures that configuration excludes incompatible credential providers
```

Comment:
explicitly patch the config before line 161 to ensure it is good everywhere, so no need to call `setConf()` again. This is what is done elsewhere

Reply:
Done.

Note: there is 2x `excludeIncompatibleCredentialProviders` calls, because the AbfsConfiguration does it again. Another approach would be to move the `super.initialize` after the `AbfsConfiguration` is ready, there is a v1 patch uploaded to this PR, check that version too. The v2 is simpler and it's less error prone.